### ISSUE TITLE
what's the different between sidecar and javaAgent?

### ISSUE THREAD

**Issue by zemochen** [megaease/easemesh#137]
i was wonder that did agent can replace sidecar? or sidecar replace agent.
i mean weather if i can only deploy one of them ?

**Comment by xxx7xxxx** [comment]
As you can see in the architecture[1], the sidecar and agent cooperate to complete kinds of tasks. Sidecar would communicate with mesh control plane to get governance information. On one hand, it could govern apps written in any language. On the other hand, it couldn't reach deeper observability. But EaseAgent uses bytecode technique in Java ecosystem to complete the left part in terms of our requirements.

So in conclusion sidecar and agent collaborate to reach EaseMesh's goals.

[1] https://github.com/megaease/easemesh#3-architecture
